tests/e2e/asserts_test.go

```diff
@@ -178,6 +178,47 @@ func AssertSwitchoverWithHistory(
 				return nil
 			}, timeout).ShouldNot(HaveOccurred())
 		})
+	} else {
+		By("verifying that the all standbys are streaming from the new primary", func() {
+			timeout := 120
+			commandTimeout := time.Second * 10
+			Eventually(func(g Gomega) {
+				podList, err := env.GetClusterPodList(namespace, clusterName)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(podList.Items).To(HaveLen(oldPodListLength))
+				var standbys []interface{}
+				for _, pod := range podList.Items {
+					if specs.IsPodStandby(pod) {
+						standbys = append(standbys, pod.Name)
+					}
+				}
+
+				for _, pod := range podList.Items {
+					stdout, _, err := env.ExecCommand(
+						env.Ctx,
+						pod,
+						specs.PostgresContainerName,
+						&commandTimeout,
+						"psql", "-U", "postgres", "-tAc",
+						"select string_agg(application_name, ',')  from pg_stat_replication;",
+					)
+					g.Expect(err).ToNot(HaveOccurred())
+
+					if specs.IsPrimary(pod.ObjectMeta) {
+						appNames := strings.Split(strings.TrimSpace(stdout), ",")
+						g.Expect(appNames).To(
+							ContainElements(standbys...),
+							"not all standbys are streaming from the new primary "+pod.Name,
+						)
+					} else {
+						g.Expect(strings.TrimSpace(stdout)).To(
+							BeEmpty(),
+							fmt.Sprintf("the standby %s should not stream to any other instance", pod.Name),
+						)
+					}
+				}
+			}, timeout).ShouldNot(HaveOccurred())
+		})
```

Comment:
Hello @YanniHu1996, what are we trying to achieve here? 

I've added line 188, because the list could have returned a different number of instances, and the test would have still succeeded. 

cc @leonardoce if you can help me understand given that you wrote the patch and have better insights

Reply:
I think Yanni is trying to assess that:

1. the HA replicas are connected to the designed primary and streaming WALs from there
2. the HA replicas are not streaming to anyone else
3. the designed primary is streaming from its source

We don't actually test the actual topology as we don't cover the third point.
BTW: why don't we use the cluster definition? We really need just the pod names and the name of the current and designated primary.

Reply:
Thanks for your comments, I submitted the commit based on them

https://github.com/cloudnative-pg/cloudnative-pg/pull/5209/commits/8c657c743598816b9a048cfc0b510d06c2200683